connmgr/error.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package connmgr
+
+type Err string
+
+func (e Err) Error() string { return string(e) }
+
+type Error struct {
+	Description string
+	Err         error
+}
+
+func (e Error) Error() string { return e.Description }
+func (e Error) Unwrap() error { return e.Err }
+
+var (
+	// ErrDialNil is used to indicate that Dial cannot be nil in
+	// the configuration.
+	ErrDialNil = Error{"dial cannot be nil", Err("ErrDialNil")}
```

Comment:
This isn't following what the error handling issue suggested and does not have the properties it intends to provide as a result since you can no longer do `errors.Is(err, connmgr.ErrDialNil)` with unwrapping and all.

It should be:
```Go
ErrDialNil = Err("ErrDialNil")
```

Then, when returning errors, it return an `Error` that adds the description **with context**.